lib/data/datastore/weather_datastore.dart

```diff
@@ -18,9 +18,17 @@ class WeatherDatastore {
   final YumemiWeather weatherClient;
 
   Map<String, dynamic> getWeather(WeatherForecastTarget target) {
-    final json = jsonEncode(target);
+    final json = toJson(target);
     final weatherJsonData = weatherClient.fetchWeather(json);
-    final weatherData = jsonDecode(weatherJsonData) as Map<String, dynamic>;
+    final weatherData = toMap(weatherJsonData);
     return weatherData;
   }
+
+  String toJson(WeatherForecastTarget target) {
+    return jsonEncode(target);
+  }
+
+  Map<String, dynamic> toMap(String json) {
+    return jsonDecode(json) as Map<String, dynamic>;
+  }
```

Comment:
[nits]
おそらくファイル内とテスト以外では使用しないと思いますので、 `@visibleForTesting` をつけておくとより安全に開発進められそうです。

```suggestion
  @visibleForTesting
  String toJson(WeatherForecastTarget target) {
    return jsonEncode(target);
  }

  @visibleForTesting
  Map<String, dynamic> toMap(String json) {
    return jsonDecode(json) as Map<String, dynamic>;
  }
```


[FYI]
https://pub.dev/packages/meta

`freezed_annotation` をインポートしても使えます 🙆‍♂️ 
```dart
import 'package:freezed_annotation/freezed_annotation.dart';
```

Reply:
確かに`@visibleForTeting`アノテーションをつけた方が安全に開発できますね...!
ご教示ありがとうございます！こちらで修正しました。
28db83cd7bc74d08c7916388f0b2aa17105d677a

また、@visibleForTestingだけだと警告のみの表示で無視することもできてしまうので、analyzerのerrorに設定しました。